src/items/functions.md

```diff
@@ -8,7 +8,7 @@
 > &nbsp;&nbsp; &nbsp;&nbsp; [_BlockExpression_]
 >
 > _FunctionQualifiers_ :\
-> &nbsp;&nbsp; `const`<sup>?</sup> `unsafe`<sup>?</sup> (`extern` _Abi_<sup>?</sup>)<sup>?</sup>
+> &nbsp;&nbsp; `async`<sup>?</sup> `const`<sup>?</sup> `unsafe`<sup>?</sup> (`extern` _Abi_<sup>?</sup>)<sup>?</sup>
```

Comment:
This grammar is wrong; `async const fn` will not parse.

Reply:
oh yeah I meant to check on that, thanks

Reply:
ugh, it's going to be a pain then. It feels like we *should* parse this (and reject later).

Reply:
Maybe... but that also entails deciding on an order, which I would prefer not to do at this juncture.

For now you can add a new production `effect_qualifier ::= "const" | "async"` and then use that one. That doesn't seem too cumbersome.

(The extern bit is also wrong and should be added to `effect_qualifier`)

Reply:
Tho `unsafe extern "C" fn f() {}` is valid syntax which does add a bit of pain.

Reply:
> but that also entails deciding on an order, which I would prefer not to do at this juncture.

I won't debate the point, but since it's an error if they occur together, it doesn't really. =) 

(I think I personally would prefer to accept qualifiers in any order, and then sort them in some canonical order with rustfmt, but anyway this PR just aims to document what's there.)

I introduced `AsyncConstQualifier` for now.

Reply:
> I won't debate the point, but since it's an error if they occur together, it doesn't really. =)

(It matters wrt. `#[cfg(FALSE)]` which in turn creates a distinction between syntax and semantics (AST and HIR) that matters wrt. stability guarantees)

> (I think I personally would prefer to accept qualifiers in any order, and then sort them in some canonical order with rustfmt, but anyway this PR just aims to document what's there.)

(Yeah maybe; It ties into the question of effect polymorphism and such... but for now I am mainly in favor of procrastination ^^)

> I introduced `AsyncConstQualifier` for now.

Thanks!